python/mmSolver/tools/screenspacerigbake/tool.py

```diff
@@ -0,0 +1,803 @@
+                nodes_to_delete.append(name + "_screenZMaster")
+                nodes_to_delete.append(name + "_screenXYMaster")
+
+                nodes_list = cmds.listConnections(name + "_screenZMaster")
+                for node in nodes_list:
+                    if "screenx_condition" in node:
+                        nodes_to_delete.append(node)
+                    if "screeny_condition" in node:
+                        nodes_to_delete.append(node)
+
+                animLayer = rigName + "Layer"
+                if cmds.animLayer(animLayer, query=True, exists=True):
+                    nodes_to_delete.append(animLayer)
+
+        # Delete selected nodes
+        nodes_to_delete = [n for n in nodes_to_delete if cmds.objExists(n)]
+        cmds.delete(nodes_to_delete)
+        self.refreshRigsList()
+        return
+
+    def lockUnlockAttributes(self, tfm_node, lock=True):
+        attr_list = TRANSFORM_ATTR_LIST
+        for attr in attr_list:
+            cmds.setAttr(tfm_node + "." + attr, lock=lock)
+
+    def Match(self):
+        selected_items = self.rigs_list.selectedItems()
+        if len(selected_items) < 2:
+            _display_warning_ui("at least two rigs must"
+                                " be selected from Rigs list.")
+            return
+
+        # Get parent
+        selected_parent = selected_items[0]
+        parent_main = selected_parent.text(0)
+        parent_base_name = parent_main.split("_screenSpaceRig")[0]
+        parent = parent_base_name + "_screenZMaster"
+
+        try:
+            self.lockUnlockAttributes(parent, lock=False)
+
+            selected_children = selected_items[1:]
+            for selected_child in selected_children:
+                child_main = selected_child.text(0)
+
+                child_base_name = child_main.split("_screenSpaceRig")[0]
+                child = child_base_name + "_screenZMaster"
+
+                self.lockUnlockAttributes(child, lock=False)
+
+                attr_list = TRANSFORM_ATTR_LIST
+                for attr in attr_list:
+                    plug = child + '.' + attr
+                    cmd = ('source channelBoxCommand; '
+                           'CBdeleteConnection "%s";')
+                    mel.eval(cmd % plug)
+
+                cmds.parent(child, parent)
+                cmds.parent(child_main, parent_main)
+                cmds.cutKey(child_main, cl=True, at=SCREEN_Z_DEPTH_ATTR_NAME)
+
+                child_scale = cmds.getAttr(child + ".sx")
+                cmds.setAttr(child_main + ".screenzdepth", child_scale)
+
+                src = child_main + "." + SCREEN_Z_DEPTH_ATTR_NAME
+                cmds.connectAttr(src, child + ".scaleX", f=True)
+                cmds.connectAttr(src, child + ".scaleY", f=True)
+                cmds.connectAttr(src, child + ".scaleZ", f=True)
+
+                self.lockUnlockAttributes(child, lock=True)
+            self.lockUnlockAttributes(parent, lock=True)
+        except:
```

Comment:
It would be good to remove the need for a try/except unless there is a good reason for it's use.
If there is a good reason, can you write a comment explaining why it's needed?